Foundation/NSFileManager.swift

```diff
@@ -842,6 +842,44 @@ public class NSDirectoryEnumerator : NSEnumerator {
     }
 }
 
+internal class NSPathDirectoryEnumerator: NSDirectoryEnumerator {
+    let baseURL: NSURL
+    let innerEnumerator : NSDirectoryEnumerator
+    override var fileAttributes: [String : AnyObject]? {
+        NSUnimplemented()
+    }
+    override var directoryAttributes: [String : AnyObject]? {
+        NSUnimplemented()
+    }
+    
+    override var level: Int {
+        NSUnimplemented()
+    }
+    
+    override func skipDescendants() {
+        NSUnimplemented()
+    }
+    
+    init?(path: String) {
+        let url = NSURL(fileURLWithPath: path)
+        self.baseURL = url
+        guard let ie = NSFileManager.defaultManager().enumeratorAtURL(url, includingPropertiesForKeys: nil, options: NSDirectoryEnumerationOptions(), errorHandler: nil) else {
+            return nil
+        }
+        self.innerEnumerator = ie
+    }
+    
+    override func nextObject() -> AnyObject? {
+        let o = innerEnumerator.nextObject()
+        guard let url = o as? NSURL else {
+            return nil
+        }
+        let path = url.path!.stringByReplacingOccurrencesOfString(baseURL.path!+"/", withString: "")
```

Comment:
perhaps it might be safer to funnel to lastPathComponent instead of doing string manipulation?